src/main/java/net/glowstone/net/GlowBufUtils.java

```diff
@@ -127,59 +131,68 @@ public static void writeMetadata(ByteBuf buf, List<Entry> entries) throws IOExce
                 }
             }
 
-            switch (index.getType()) {
-                case BYTE:
-                    buf.writeByte((Byte) value);
-                    break;
-                case INT:
-                    ByteBufUtils.writeVarInt(buf, (Integer) value);
-                    break;
-                case FLOAT:
-                    buf.writeFloat((Float) value);
-                    break;
-                case STRING:
-                    ByteBufUtils.writeUTF8(buf, (String) value);
-                    break;
-                case CHAT:
-                    writeChat(buf, (TextMessage) value);
-                    break;
-                case ITEM:
-                    writeSlot(buf, (ItemStack) value);
-                    break;
-                case BOOLEAN:
-                    buf.writeBoolean((Boolean) value);
-                    break;
-                case VECTOR:
-                    EulerAngle angle = (EulerAngle) value;
-                    buf.writeFloat((float) Math.toDegrees(angle.getX()));
-                    buf.writeFloat((float) Math.toDegrees(angle.getY()));
-                    buf.writeFloat((float) Math.toDegrees(angle.getZ()));
-                    break;
-                case POSITION:
-                case OPTPOSITION:
-                    BlockVector vector = (BlockVector) value;
-                    buf.writeLong(Position.getPosition(vector));
-                    break;
-                case DIRECTION:
-                    ByteBufUtils.writeVarInt(buf, (Integer) value);
-                    break;
-                case OPTUUID:
-                    writeUuid(buf, (UUID) value);
-                    break;
-                case BLOCKID:
-                    ByteBufUtils.writeVarInt(buf, (Integer) value);
-                    break;
-                case NBTTAG:
-                    writeCompound(buf, (CompoundTag) value);
-                    break;
-                default:
-                    // do nothing
-            }
+            writeValue(buf, value, index.getType());
         }
 
         buf.writeByte(0xff);
     }
 
+    private static void writeValue(ByteBuf buf, Object value, MetadataType type) throws IOException {
+        switch (type) {
+            case BYTE:
+                buf.writeByte((Byte) value);
+                break;
+            case INT:
+                ByteBufUtils.writeVarInt(buf, (Integer) value);
+                break;
+            case FLOAT:
+                buf.writeFloat((Float) value);
+                break;
+            case STRING:
+                ByteBufUtils.writeUTF8(buf, (String) value);
+                break;
+            case CHAT:
+                writeChat(buf, (TextMessage) value);
+                break;
+            case OPTCHAT:
+                writeChat(buf, (TextMessage) value);
```

Comment:
Missing a break. Merge with previous case.

Reply:
????

Reply:
What are you question marking?